### PR TITLE
Encrypted library version 3: use different salt for each library.

### PR DESCRIPTION
**Pull request by caixiangyue**

### CODE REVIEW

**Comment by christmart** [comment]
Please comment in English.

File: daemon/clone-mgr.c [hunk 1]
```diff
@@ -1298,8 +1325,23 @@ seaf_clone_manager_add_download_task (SeafCloneManager *mgr,
...
+         json_t *object = NULL;
+ 
+         object = json_loads (more_info, 0, &jerror);
+         if (!object) {
+             seaf_warning ("Failed to load more sync info from json: %s.\n", jerror.text);
+             return NULL;
+         }
+         json_t *string = json_object_get (object, "repo_salt");
+         if (string)
+             repo_salt = g_strdup (json_string_value (string));
```
Comment (killing) [inline daemon/clone-mgr.c]: repo_salt is not free() later.

File: daemon/clone-mgr.c [hunk 2]
```diff
@@ -1150,25 +1162,41 @@ seaf_clone_manager_add_task (SeafCloneManager *mgr,
     char *worktree;
     char *ret;
     gboolean sync_wt_name = FALSE;
+    char *repo_salt;
```
Comment (killing) [inline daemon/clone-mgr.c]: These variables need to be initialized to NULL.

File: daemon/clone-mgr.c [hunk 3]
```diff
@@ -1150,25 +1162,41 @@ seaf_clone_manager_add_task (SeafCloneManager *mgr,
...
+        goto out;
     }
 #endif
 
-    if (passwd &&
-        !check_encryption_args (magic, enc_version, random_key, error))
-        return NULL;
+    if (more_info) {
+        json_error_t jerror;
+        json_t *object = NULL;
```
Comment (killing) [inline daemon/clone-mgr.c]: It's not necessary to set object to NULL, since json_loads will return NULL on failure.

File: daemon/clone-mgr.c [hunk 4]
```diff
@@ -1236,7 +1264,12 @@ seaf_clone_manager_add_task (SeafCloneManager *mgr,
                            email, more_info,
                            sync_wt_name,
                            error);
-    g_free (worktree);
+
+out:
+    if (worktree)
```
Comment (killing) [inline daemon/clone-mgr.c]: You don't need to check whether a variable is NULL before g_free(). g_free() will check it.

**Pull request merged**